tasks/quality_gates.py

```diff
@@ -0,0 +1,144 @@
+
+    pr_commenter(ctx, title=title, body=body)
+
+
+def _print_quality_gates_report(gate_states: list[dict[str, typing.Any]]):
+    print(color_message("======== Static Quality Gates Report ========", "magenta"))
+    for gate in sorted(gate_states, key=lambda x: x["error_type"] is not None):
+        if gate["error_type"] is None:
+            print(color_message(f"Gate {gate['name']} succeeded {SUCCESS_CHAR}", "blue"))
+        elif gate["error_type"] == "AssertionError":
+            print(
+                color_message(
+                    f"Gate {gate['name']} failed {FAIL_CHAR} because of the following assertion failures :\n{gate['message']}",
+                    "orange",
+                )
+            )
+        else:
+            print(
+                color_message(
+                    f"Gate {gate['name']} failed {FAIL_CHAR} with the following stack trace :\n{gate['message']}",
+                    "orange",
+                )
+            )
+
+
+@task
+def parse_and_trigger_gates(ctx, config_path="test/static/static_quality_gates.yml"):
+    """
+    Parse and executes static quality gates
+    :param ctx: Invoke context
+    :param config_path: Static quality gates configuration file path
+    :return:
+    """
+    with open(config_path) as file:
+        config = yaml.safe_load(file)
+
+    gate_list = list(config.keys())
+    quality_gates_mod = __import__("tasks.static_quality_gates", fromlist=gate_list)
+    print(f"{config_path} correctly parsed !")
+    metric_handler = GateMetricHandler(
+        git_ref=os.environ["CI_COMMIT_REF_SLUG"], bucket_branch=os.environ["BUCKET_BRANCH"]
+    )
+
+    print(f"The following gates are going to run:\n\t- {"\n\t- ".join(gate_list)}")
+    final_state = True
+    gate_states = []
+    for gate in gate_list:
+        gate_inputs = config[gate]
+        gate_inputs["ctx"] = ctx
+        gate_inputs["metricHandler"] = metric_handler
+        try:
+            gate_mod = getattr(quality_gates_mod, gate)
+            gate_mod.entrypoint(**gate_inputs)
+            print(f"Gate {gate} succeeded !")
+            gate_states.append({"name": gate, "state": True, "error_type": None, "message": None})
+        except AssertionError as e:
+            print(f"Gate {gate} failed ! (AssertionError)")
+            final_state = False
+            gate_states.append({"name": gate, "state": False, "error_type": "AssertionError", "message": str(e)})
+        except Exception:
+            print(f"Gate {gate} failed ! (StackTrace)")
+            final_state = False
+            gate_states.append(
+                {"name": gate, "state": False, "error_type": "StackTrace", "message": traceback.format_exc()}
+            )
+    if not final_state:
+        ctx.run("datadog-ci tag --level job --tags static_quality_gates:\"failure\"")
+    else:
+        ctx.run("datadog-ci tag --level job --tags static_quality_gates:\"success\"")
```

Comment:
```suggestion
    ctx.run(f"datadog-ci tag --level job --tags static_quality_gates:\"{final_state}\"")
```
Alternative solution to the above: you keep `final_state` but you initialize it with `final_state = "success"` and you set it to `failure` on your `except` instructions. In this way the meaning of `final_state` is correct and you save yourself an `if` condition. (I think I prefer this one)

Reply:
Agreed 👍